project_task_id_in_display_name/views/project_task_views.xml

```diff
@@ -0,0 +1,77 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<odoo>
+
+    <record id="view_task_form2" model="ir.ui.view">
+        <field name="name">Task Form With Display Name In Read Mode</field>
+        <field name="model">project.task</field>
+        <field name="inherit_id" ref="project.view_task_form2"/>
+        <field name="arch" type="xml">
+            <field name="name" position="before">
+                <field name="display_name" class="oe_read_only"/>
+            </field>
+            <field name="name" position="attributes">
+                <attribute name="class">oe_edit_only</attribute>
+            </field>
+        </field>
+    </record>
+
+    <record id="view_task_kanban" model="ir.ui.view">
+        <field name="name">Task Kanban With Display Name Instead Of Name</field>
+        <field name="model">project.task</field>
+        <field name="inherit_id" ref="project.view_task_kanban"/>
+        <field name="arch" type="xml">
+            <xpath expr="//strong[hasclass('o_kanban_record_title')]/s/field[@name='name']" position="before">
+                <field name="display_name"/>
+            </xpath>
+            <xpath expr="//strong[hasclass('o_kanban_record_title')]/s/field[@name='name']" position="attributes">
+                <attribute name="invisible">1</attribute>
+            </xpath>
+            <xpath expr="//strong[hasclass('o_kanban_record_title')]/t/field[@name='name']" position="before">
+                <field name="display_name"/>
+            </xpath>
+            <xpath expr="//strong[hasclass('o_kanban_record_title')]/t/field[@name='name']" position="attributes">
+                <attribute name="invisible">1</attribute>
+            </xpath>
+        </field>
+    </record>
+
+    <record id="view_task_tree2" model="ir.ui.view">
+        <field name="name">Task List: Display Name Instead Of Name</field>
+        <field name="model">project.task</field>
+        <field name="inherit_id" ref="project.view_task_tree2"/>
+        <field name="arch" type="xml">
+            <field name="name" position="before">
+                <field name="display_name"/>
+            </field>
+            <field name="name" position="attributes">
```

Comment:
Hello @rivo2302 

In the original views we have a new widget **name_with_subtask_count** added to **name** fields
we replace in this module **name** with **display_name**, should we keep the widget?

[See code here](https://github.com/odoo/odoo/blob/16.0/addons/project/views/project_views.xml#L1516)

Reply:
@majouda 

Yes, we are slightly altering the native behavior, but I think we should keep the module as it is. The name_with_subtask_count widget only appends the number of subtasks to the task name, so it doesn't significantly impact the functionality.


![image](https://github.com/user-attachments/assets/899ebc26-133d-4246-99f0-ae9f478fa8f1)
![image](https://github.com/user-attachments/assets/c4d9ce50-07c8-4eb6-9dca-0365475e2022)